internal/pkg/deploy/deploy.go

```diff
@@ -121,25 +121,47 @@ func (s *Store) ListEnvironmentsDeployedTo(appName string, svcName string) ([]st
 	if err != nil {
 		return nil, fmt.Errorf("list environment for app %s: %w", appName, err)
 	}
-	var envsWithDeployment []string
+	type result struct {
+		name string
+		err  error
+	}
+	deployedEnv := make(chan result, len(envs))
```

Comment:
`defer close(deployedEnv)` instead of closing after looping, because in case of an error it won't get closed